Add DoesDocumentExists check for stress tests

Harivallabha-Rangarajan opened the pull request:
- Add explicit check DoesDocumentExists for stress tests, to check if the document exists.
- Previously, we just assume that the document exists when a testId is supplied to nodeStressTest.ts; This change ensures that we explicitly check if a document corresponding to the testId passed does indeed exist.
- Implemented for odsp driver, raises "not implemented" for other drivers.
- Modify type validation checks to take into account the forward compatibility that will be broken for certain classes and functions because of this change (addition of the doesDocumentExists method to ITestDriver).


EDIT:

- Prior to this change, we just assume that the corresponding document exists when a testId is supplied to nodeStressTest.ts; And when such a document does not actually exist, the stress test would simply error out. Rather, what we want to do now is to check if the corresponding document exists, and in the case that it does not exist, we go ahead and create a document (which is the default behavior when no testId is passed to nodeStressTest.ts) rather than error out and stop the test. The name of the document created is the testId specified.

Also, we previously didn't support the creation of a document with the given name (we just assume a document with the given name exists, and error out if it does not. And, in case no testId is specified we used to create a document with the name set to ${Date.now().toString()}-WireFormatV1RWOptimizedSnapshot_45e4).

-  Encode the check into the function itself rather than create a separate method for it.
- Does not break the forward/backwards compatibility anymore.

anthony-murphy commented on packages/test/test-service-load/src/nodeStressTest.ts:
```diff
@@ -102,10 +102,19 @@ async function orchestratorProcess(
+    if (args.testId !== undefined) {
+        const docExists: boolean = await testDriver.doesDocumentExists(args.testId);
+        if (docExists) {
+            url = await testDriver.createContainerUrl(args.testId);
```
i don't think we need a new function. just encode this behavior in createContainerUrl and return undefined in the case where it doesn't exit

anthony-murphy commented on packages/test/test-drivers/src/routerliciousTestDriver.ts:
```diff
@@ -175,4 +175,8 @@ export class RouterliciousTestDriver implements ITestDriver {
     }
+
+    async doesDocumentExists(testId: string): Promise<boolean> {
+        throw new Error("not implemented");
```
this will regress our stress runs. it must be implemented for all drivers if it is added, which i don't think it should be

anthony-murphy commented:
why is this necessary? please open an issue that describe the scenario/need for this change

Harivallabha-Rangarajan commented:
@anthony-murphy 

Sure, I'll encode this behavior into the function itself rather than create a separate method for it - so that it does not regress other stress tests.

Prior to this change, we just assume that the corresponding document exists when a testId is supplied to nodeStressTest.ts; And when such a document does not actually exist, the stress test would simply error out. Rather, what we want to do now is to check if the corresponding document exists, and in the case that it does not exist, we go ahead and create a document (which is the default behavior when no testId is passed to nodeStressTest.ts) rather than error out and stop the test.

Harivallabha-Rangarajan commented:
Hey Tony, I can see that all the jobs in the pipeline are succeeding except for the "Check for extraneous modified files"

https://dev.azure.com/fluidframework/public/_build/results?buildId=78640&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=035f10b1-6721-511e-bd48-d7e028467fc2

Could you please help me understand what this is about? (I'm not able to see which bash script does this check for extraneous modified files, and which ones it is referring to)

Harivallabha-Rangarajan commented:
Oh yes, certainly. I did do a full local build - and it was successful. The type validation files were generated as well.

>> you can't just delete files.

I guess you're referring to commit c90e5f1 [remove type validation files that will be auto generated on build]. I had checked in a couple of auto generated type validation files in the commit just before that. I'm just removing those files in commit c90e5f1 as I suppose that the auto generated type validation files _shouldn't_ be checked in. (please correct me if my understanding is incorrect/if those files should also be checked in)

anthony-murphy commented:
> Oh yes, certainly. I did do a full local build - and it was successful. The type validation files were generated as well.
> 
> > > you can't just delete files.
> 
> I guess you're referring to commit [c90e5f1](https://github.com/microsoft/FluidFramework/commit/c90e5f1338ae3d3d7b04868ad7788c003cd33a9f) [remove type validation files that will be auto generated on build]. I had checked in a couple of auto generated type validation files in the commit just before that. I'm just removing those files in commit [c90e5f1](https://github.com/microsoft/FluidFramework/commit/c90e5f1338ae3d3d7b04868ad7788c003cd33a9f) as I suppose that the auto generated type validation files _shouldn't_ be checked in. (please correct me if my understanding is incorrect/if those files should also be checked in)

they should be checked in. you are actively deleting files that exist in the remote git repository

Harivallabha-Rangarajan committed:
use a user-specified createTestId flag instead of a check through a method call

alexvy86 commented:
Seems like something went wrong here with a merge/rebase, 48 commits and 380 files with changes is much more than the PR is about. @Harivallabha-Rangarajan maybe it is a good idea to rebase this branch off the tip of `main` so we see only the intended changes in the PR? Or start a new branch from the tip of `main` and copy the changes to that one, if the rebase is too tricky.

CraigMacomber committed:
Use interface for named brand types (#11058)

Rebase on top of latest changes - Add an explicit check for Document
corresponding to testId for stress tests

remove type validation files that will be auto generated on build

resolve rebase conflicts

lint fix

add back the type validation files

Add DoesDocumentExistsCheck for stress tests

remove type validation files that will be auto generated on build

remove the DoesDocumentExists method; encode the behavior within the createContainerUrl function instead

lint fix

add back the type validation files

Harivallabha-Rangarajan commented:
@alexvy86 You're right, yeah it's become quite messy after the rebase. I'll close this PR and recreate the changes on top of main again, the changes themselves are small, and it would be cleaner to do so :")
